File: day4-orchestration-evals-cicd/capstone/reference-solution/java/core/src/test/java/com/airamatrix/capstone/LiveTest.java
package com.airamatrix.capstone;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;
import static org.junit.jupiter.api.Assumptions.assumeTrue;

import java.util.List;

import org.junit.jupiter.api.Test;

import com.airamatrix.labkit.Config;
import com.airamatrix.labkit.GatewayClient;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.node.ObjectNode;

/**
 * One real run through the training gateway (about $0.03). Runs only with LAB_LIVE=1 - a live test costs money
 * and never runs by default. Unset AIRA_OPS_TOKEN / AIRA_OPS_APPLY_TOKEN first: the agent refuses to start with them.
 *   LAB_LIVE=1 mvn -q -f java/pom.xml -pl core -am test -Dtest=LiveTest -Dsurefire.failIfNoSpecifiedTests=false
 */
class LiveTest {
    @Test
    void theBacklogCasePassesItsGoldenChecksLive() throws Exception {
        assumeTrue("1".equals(System.getenv("LAB_LIVE")), "LAB_LIVE=1 not set - live test skipped");
        JsonNode golden = Evals.load(Repo.solution().resolve("golden/cases.json"));
        JsonNode kase = Evals.select(golden, "backlog-acc1001").get(0);
        Config cfg = new Config().require();
        try (PrivateOps ops = PrivateOps.start(List.of("ACC-1001"), false)) {
            ObjectNode r = Evals.local(ops, Evals.localAgents(new GatewayClient(cfg)::messages, cfg.model, 10), 0.30).run(kase);
            assertTrue(!r.has("error"), String.valueOf(r.path("error")));
            assertEquals("awaiting_approval", r.path("status").asText(), r.toPrettyString());
            ObjectNode g = Checks.gradeCase(kase, r);
            assertTrue(g.path("passed").asBoolean(), g.toPrettyString());
        }
    }
}

File: day4-orchestration-evals-cicd/capstone/reference-solution/node/tests/live.test.mjs
// Live: one real run through the training gateway (about $0.03). Only with LAB_LIVE=1 - a live test costs money.
//   LAB_LIVE=1 node --test day4-orchestration-evals-cicd/capstone/reference-solution/node/tests
import assert from "node:assert/strict";
import { test } from "node:test";
import { Config, GatewayClient } from "../../../../../labkit/node/agentic-core.mjs";
import { ResponderAgent, localModel } from "../lib/agent.mjs";
import { gradeCase } from "../lib/checks.mjs";
import * as evals from "../lib/evals.mjs";
import { HttpOpsReader } from "../lib/ops.mjs";
import { PrivateOps } from "../lib/private-ops.mjs";
import { solution } from "../lib/repo.mjs";
import * as responder from "../lib/responder.mjs";
import { parseInstant } from "../lib/time.mjs";
import { tracer } from "./helpers.mjs";
import path from "node:path";

test("live: the backlog case end to end passes its golden checks", { skip: process.env.LAB_LIVE !== "1" && "set LAB_LIVE=1 (costs money)" }, async () => {
  const kase = evals.select(evals.load(path.join(solution(), "golden", "cases.json")), "backlog-acc1001")[0];
  const ops = await PrivateOps.start([kase.account]);
  try {
    const cfg = new Config().require();
    const agent = new ResponderAgent(localModel(new GatewayClient(cfg)), cfg.model, 10, 0.3, () => undefined);
    const o = await responder.run("live-1", kase.account, parseInstant(kase.as_of), kase.question,
      new HttpOpsReader(ops.url, ops.readTokens[kase.account]), agent, tracer());
    const g = gradeCase(kase, o.toJson());
    assert.ok(g.passed, JSON.stringify(g.checks.filter((c) => !c.passed)));
    assert.ok(o.costUsd < 0.3, `cost ${o.costUsd}`);
  } finally {
    await ops.close();
  }
});
